packages/runtime/datastore/src/dataStoreRuntime.ts

```diff
@@ -212,6 +212,14 @@ export class FluidDataStoreRuntime extends EventEmitter implements IFluidDataSto
                         (content, localOpMetadata) => this.submitChannelOp(path, content, localOpMetadata),
                         (address: string) => this.setChannelDirty(address),
                         tree.trees[path]);
+                    // If the data store is loaded after the container is attached, then attach the channel right away.
```

Comment:
It's a bit hard to follow code (sorry I missed last changes, so feedback about existing code).

IFluidDataStoreContextType - I feel like this should be removed. IFluidDataStoreContext is casted to it on line 96, saying that isLocalDataStore should be on IFluidDataStoreContext and in current form it should be of boolean | undefined type. It would be great if it can be just boolean.

Could you please promote IF statement on line 202 above enumeration of Object.keys(tree..tree)? dataStoreContext.isLocalDataStore is constant for the loop, so it should be above it.

The code on line 203 - when do we have storage representation for local data store?
This only for Drafts de-serializaiton, right?
Would be great to add a comment on that.

Reply:
Added a comment. Tony suggested that we don't want to expose it IFluidDataStoreContext and just want to keep it internal to runtime for implementation and I agree. So the typecast is necessary and we know that the boolean will be present(not undefined) and we assign it in dataStoreContext.ts.

Reply:
This to me does not make sense. These are two different layers - container layer, and data store layer. I, as a developer, need to be able to implement my own version of DataStoreRuntime, using public API surface. Making comment that it's not piblic is equivalent to me that I can't do this work, without reverse-engineering and using undocumented API.

Maybe it needs to take another form, but data store runtime layer should not rely on anything non-public in container layer.

Reply:
But IFluidDataStoreContextType is also public it is just that it is not in main set of apis in IFluidDataStoreContext as it sounds more internal to runtime implementation. Do you suggest adding the property in IFluidDataStoreContext interface?
@vladsud @anthony-murphy ?

Reply:
The contract should be clear. I do not see an advantage of existing behavior compared to putting this property on main object as boolean | undefined. Next improvement step if we can drop undefined part.

Reply:
It is boolean. How is it undefined?

Reply:
It's undefined on IFluidDataStoreContext interface. At least that's how I'd interpret this code that cases and object without property to derive interface that has this property. I'd rather see it that way (undefined | boolean) on original interface (and no additional interfaces) if that's the intention, or better - as boolean property on IFluidDataStoreContext